packages/@aws-cdk/aws-kinesisfirehose/lib/delivery-stream.ts

```diff
@@ -0,0 +1,256 @@
+import * as cloudwatch from '@aws-cdk/aws-cloudwatch';
+import * as ec2 from '@aws-cdk/aws-ec2';
+import * as iam from '@aws-cdk/aws-iam';
+import * as cdk from '@aws-cdk/core';
+import { RegionInfo } from '@aws-cdk/region-info';
+import { Construct } from 'constructs';
+import { IDestination } from './destination';
+import { CfnDeliveryStream } from './kinesisfirehose.generated';
+
+const PUT_RECORD_ACTIONS = [
+  'firehose:PutRecord',
+  'firehose:PutRecordBatch',
+];
+
+/**
+ * Represents a Kinesis Data Firehose delivery stream.
+ */
+export interface IDeliveryStream extends cdk.IResource, iam.IGrantable, ec2.IConnectable, cdk.ITaggable {
+  /**
+   * The ARN of the delivery stream.
+   *
+   * @attribute
+   */
+  readonly deliveryStreamArn: string;
+
+  /**
+   * The name of the delivery stream.
+   *
+   * @attribute
+   */
+  readonly deliveryStreamName: string;
+
+  /**
+   * Grant the `grantee` identity permissions to perform `actions`.
+   */
+  grant(grantee: iam.IGrantable, ...actions: string[]): iam.Grant;
+
+  /**
+   * Grant the `grantee` identity permissions to perform `firehose:PutRecord` and `firehose:PutRecordBatch` actions on this delivery stream.
+   */
+  grantPutRecords(grantee: iam.IGrantable): iam.Grant;
+
+  /**
+   * Return the given named metric for this delivery stream.
+   */
+  metric(metricName: string, props?: cloudwatch.MetricOptions): cloudwatch.Metric;
+}
+
+/**
+ * Base class for new and imported Kinesis Data Firehose delivery streams.
+ */
+export abstract class DeliveryStreamBase extends cdk.Resource implements IDeliveryStream {
+
+  public abstract readonly deliveryStreamName: string;
+
+  public abstract readonly deliveryStreamArn: string;
+
+  public abstract readonly grantPrincipal: iam.IPrincipal;
+
+  /**
+   * Network connections between Kinesis Data Firehose and other resources, i.e. Redshift cluster.
+   */
+  public readonly connections: ec2.Connections;
+
+  public readonly tags = new cdk.TagManager(cdk.TagType.STANDARD, 'AWS::KinesisFirehose::DeliveryStream');
```

Comment:
Yeah fairly sure tagging support should happen automatically.

Reply:
Okay thanks, updated to remove this and also removed the fact that `IDeliveryStream` implements `ITaggable`. Since `Portfolio` doesn't implement `ITaggable`. Should that interface be deprecated if it's not needed anymore? what is the context there?